### PR TITLE
PrescriptionIdentifier and dispenseridentifier attribute additions

### PR DESCRIPTION
**Pull request by mayerm94**
Add prescriptionIdentifier and dispenserIdentifier to medicationOrder and medicationDispensed

Pull requests into CQL QDM Patient API require the following. Submitter and reviewer should :white_check_mark: when done. For items that are not-applicable, note it's not-applicable ("N/A") and :white_check_mark:.

**Submitter:**
- [x] This pull request describes why these changes were made.
- [x] This PR is into the correct branch.
- [x] JIRA ticket for this PR: https://jira.mitre.org/browse/BONNIE-1703
- [x] JIRA ticket links to this PR
- [x] Code diff has been done and been reviewed (it **does not** contain: additional white space, not applicable code changes, debug statements, etc.)
- [x] Tests are included and test edge cases
- [x] Tests have been run locally and pass (remember to update Gemfile when applicable)
- [x] Code coverage has not gone down and all code touched or added is covered.
    * In rare situations, this may not be possible or applicable to a PR. In those situations:
        1. Note why this could not be done or is not applicable here:
        2. Add TODOs in the code noting that it requires a test
        3. Add a JIRA task to add the test and link it here:
- [x] Automated regression test(s) pass NA

If JIRA tests were used to supplement or replace automated tests:
- [x] JIRA test links: NA
- [x] Justification for using JIRA tests: NA
- [x] JIRA tests have been added to sprint NA

**Reviewer 1:**

Name: @hossenlopp
- [x] Code is maintainable and reusable, reuses existing code and infrastructure where appropriate, and accomplishes the task’s purpose
- [x] The tests appropriately test the new code, including edge cases
- [x] You have tried to break the code

If JIRA tests were used to supplement or replace automated tests: N/A
- [x] JIRA tests have been run and pass
- [x] You agree with the justification for use of JIRA tests or have provided input on why you disagree

**Reviewer 2:**

Name: @daco101
- [x] Code is maintainable and reusable, reuses existing code and infrastructure where appropriate, and accomplishes the task’s purpose
- [x] The tests appropriately test the new code, including edge cases
- [x] You have tried to break the code

If JIRA tests were used to supplement or replace automated tests:
- [x] JIRA tests have been run and pass
- [x] You agree with the justification for use of JIRA tests or have provided input on why you disagree

### CODE REVIEW

File: app/assets/javascripts/datatypes/medication.js.coffee [hunk 1]
```diff
@@ -282,6 +284,15 @@ class CQL_QDM.MedicationDispensed extends CQL_QDM.QDMDatatype
   authorDatetime: ->
     @_authorDatetime
 
+  ###
+  @returns {Id}
+  ###
+  dispenserIdentifier: ->
```
Comment (hossenlopp) [inline app/assets/javascripts/datatypes/medication.js.coffee]: Should be `dispenserId` to match QDM.

File: app/assets/javascripts/datatypes/medication.js.coffee [hunk 2]
```diff
@@ -282,6 +284,15 @@ class CQL_QDM.MedicationDispensed extends CQL_QDM.QDMDatatype
...
+   if @_prescriberIdentifier?
```
Comment (hossenlopp) [inline app/assets/javascripts/datatypes/medication.js.coffee]: Typo? Should be looking at `@_dispenserIdentifier`.

File: app/assets/javascripts/datatypes/medication.js.coffee [hunk 3]
```diff
@@ -312,6 +323,15 @@ class CQL_QDM.MedicationDispensed extends CQL_QDM.QDMDatatype
     else
       null
 
+  ###
+  @returns {Id}
+  ###
+  prescriberIdentifier: ->
```
Comment (hossenlopp) [inline app/assets/javascripts/datatypes/medication.js.coffee]: Should be `prescriberId` to match QDM.

File: app/assets/javascripts/datatypes/medication.js.coffee [hunk 4]
```diff
@@ -424,6 +445,15 @@ class CQL_QDM.MedicationOrder extends CQL_QDM.QDMDatatype
...
```
Comment (hossenlopp) [inline app/assets/javascripts/datatypes/medication.js.coffee]: Should be prescriberId to match QDM.

**Comment by hossenlopp** [comment]
There is no test for when dispenserId is null. Other than that it looks good.
